### PR TITLE
jlink.sh: remove duplicated check

### PR DESCRIPTION
**Pull request by kYc0o**
### Contribution description

Currently, jlink.sh makes double check for the existance of an `HEXFILE`, one of this checks  is not necessary, as it already exists a function for checking it.

### Testing procedure

Flash a board using the jlink debugger/programmer.

### CODE REVIEW

File: dist/tools/jlink/jlink.sh [hunk 1]
```diff
@@ -99,6 +93,8 @@ test_hexfile() {
         echo "Error: Unable to locate HEXFILE"
         echo "       (${HEXFILE})"
         exit 1
+    else
```
Comment (basilfx) [inline dist/tools/jlink/jlink.sh]: Do we need to print on success? If so, then `test_elffile` should also be adapted.

**Comment by basilfx** [comment]
Tested this on SLSTK3402a that uses JLink. Works fine.

File: dist/tools/jlink/jlink.sh (hunk 1 above)
Comment (kYc0o) [reply dist/tools/jlink/jlink.sh]: True, I'd say we remove that part, is not really needed. I'll remove it and squash directly.

Comment (kYc0o) [reply dist/tools/jlink/jlink.sh]: Done.

**Comment by kYc0o** [comment]
Thanks @basilfx !

**Pull request merged**